semanticdb/metacp/src/main/scala/scala/meta/internal/metacp/Main.scala

```diff
@@ -447,28 +404,58 @@ object Main {
     }
   }
 
-  private implicit class SymbolOps(sym: SymbolInfoSymbol) {
-    def isModuleClass = sym.isInstanceOf[ClassSymbol] && sym.isModule
-    def isType = sym.isInstanceOf[TypeSymbol]
-    def isClassConstructor = {
+  private implicit class NameOps(name: String) {
+    def encoded: String = {
+      if (name.isEmpty) {
+        sys.error(s"unsupported name")
+      } else {
+        val (start, parts) = (name.head, name.tail)
+        val isStartOk = Character.isJavaIdentifierStart(start)
+        val isPartsOk = parts.forall(Character.isJavaIdentifierPart)
+        if (isStartOk && isPartsOk) name
+        else "`" + name + "`"
+      }
+    }
+  }
+
+  private implicit class SymbolOps(sym: Symbol) {
+    def isModuleClass: Boolean = sym.isInstanceOf[ClassSymbol] && sym.isModule
+    def isType: Boolean = sym.isInstanceOf[TypeSymbol]
+    def isClassConstructor: Boolean = {
       sym.parent match {
         case Some(parent: ClassSymbol) if !parent.isTrait && !parent.isModule =>
           sym.name == "<init>"
         case _ =>
           false
       }
     }
+    def descriptor: String = {
+      sym match {
+        case sym: SymbolInfoSymbol => sym.infoType.descriptor
+        case sym => sys.error(s"unsupported symbol $sym")
+      }
+    }
+    def disambiguator: String = {
+      val kindred = sym.parent.get.children.filter(other => skind(other) == skind(sym))
+      val siblings = kindred.filter(_.name == sym.name)
+      val synonyms = siblings.filter(_.descriptor == sym.descriptor)
```

Comment:
Do we have any ordering guarantees on `Symbol.children`? What is the chance that `+1` and `+2` get mixed up?

Reply:
To the best of my knowledge, pickling preserves the order of symbols in scopes. Scalac namer enters symbols in the order of their appearance in source code. Also see https://github.com/retronym/scalac-stability/issues/1.